hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -1254,8 +1252,8 @@ private boolean hasOnGoingMigrationMaster(Level level) {
 
     @Override
     public boolean hasOnGoingMigrationLocal() {
-        return !activeMigrations.isEmpty() || !migrationQueue.isEmpty()
-                || migrationThread.isMigrating();
+        return !activeMigrations.isEmpty() || migrationQueue.isNonEmpty()
+                || migrationQueue.hasMigrations();
```

Comment:
why the complex query? The MigrationQueue knows if there is a MigrationTask pending or executing. So we does activeMigrations and migrationQueue no empty be used?

So why not just:

```
 public boolean hasOnGoingMigrationLocal() {return migrationQueue.hasMigrations();}
```